src/OpenTelemetry.Exporter.Prometheus/Implementation/PrometheusSerializerExt.cs

```diff
@@ -70,11 +70,25 @@ public static int WriteMetric(byte[] buffer, int cursor, Metric metric)
                     // for each MetricPoint
                     if (((int)metric.MetricType & 0b_0000_1111) == 0x0a /* I8 */)
                     {
-                        cursor = WriteLong(buffer, cursor, metricPoint.LongValue);
+                        if (metric.MetricType.IsSum())
+                        {
+                            cursor = WriteLong(buffer, cursor, metricPoint.GetLongSum());
+                        }
+                        else
+                        {
```

Comment:
It seems we do this code in a few places now (if sum - GetLongSum..). Maybe we could use an extension method on metricPoint that accepts the `MetricType`?

Reply:
These checks be avoided. We have a TODO in line 68 about this.